### DISCUSSION TITLE
Stub file?

### DISCUSSION THREAD

**Discussion by BackSeat** [helloflask/bootstrap-flask#309] [General]
Is there a stub file for bootstrap-flask? Currently mypy reports "Skipping analyzing "flask_bootstrap": module is installed, but missing library stubs or py.typed marker"

bootstrap-flask 2.3.0 installed.

**Comment by PanderMusubi** [comment]
Not at the moment, but feel free to contribute one.

**Comment by PanderMusubi** [comment]
Do you want to give it a try? See e.g. https://mypy.readthedocs.io/en/stable/stubs.html

**Comment by PanderMusubi** [comment]
@greyli what do you think about adding type hints to https://github.com/helloflask/bootstrap-flask/blob/master/flask_bootstrap/__init__.py ? Is also recommended by mypy.

**Comment by PanderMusubi** [comment]
~@greyli here is a start of adding type hints https://github.com/PanderMusubi/bootstrap-flask/commit/75c4458b45e824661e93cfe1afb00e2ff3caaa35 What do you think about it?~ (See WIP PR below)

**Comment by PanderMusubi** [comment]
Also practical:
- https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
- https://docs.python.org/3.11/library/typing.html
- https://peps.python.org/pep-0561/#packaging-type-information

**Comment by PanderMusubi** [comment]
From now one, please see this WIP PR https://github.com/helloflask/bootstrap-flask/pull/323 However, with Python 3.8 and 3.9, it could take a while before we can start to use this.
